plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/FileSystemSpoolingManager.java

```diff
@@ -127,16 +131,16 @@ public SpooledLocation location(SpooledSegmentHandle handle)
     {
         // Identifier layout:
         //
-        // ttl: long
-        // nameLength: int
-        // name: byte[]
+        // ulid: byte[128]
+        // queryIdLength: byte
+        // queryId: string
         // hasEncryptionKey: boolean
         // encryptionKeyHash: long
         FileSystemSpooledSegmentHandle fileHandle = (FileSystemSpooledSegmentHandle) handle;
         DynamicSliceOutput output = new DynamicSliceOutput(64);
-        output.writeLong(fileHandle.validUntil().toEpochMilli());
-        output.writeInt(fileHandle.name().length());
-        output.writeBytes(utf8Slice(fileHandle.name()));
+        output.writeBytes(fileHandle.ulid());
+        output.writeShort(fileHandle.queryId().toString().length());
+        output.writeBytes(utf8Slice(fileHandle.queryId().toString()));
         output.writeBoolean(fileHandle.encryptionKey().isPresent());
         if (fileHandle.encryptionKey().isPresent()) {
             output.writeLong(XxHash64.hash(fileHandle.encryptionKey().orElseThrow()));
```

Comment:
I don't know if this is safe. xxHash is not cryptographic, so this might leak some bits from the encryption key, which could reduce the effective key strength, or otherwise leak potentially useful information to an attacker.